tests/cross_validation/c_harness/harness_sweep.c

```
@@ -0,0 +1,127 @@
+
+  FILE *cf = fopen(argv[3], "r");
+  if (!cf) { fprintf(stderr, "cannot open %s\n", argv[3]); return 5; }
+
+  /* count rows */
+  char line[1024]; UINT4 n_sets = 0;
+  while (fgets(line, sizeof(line), cf)) if (line[0] != '\n' && line[0] != '#') n_sets++;
+  rewind(cf);
+
+  UINT4 n_samples = (UINT4)ceil(FS * DURATION);
+  FILE *of = fopen(argv[4], "wb");
+  fwrite(&n_sets, sizeof(UINT4), 1, of);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)